icelake/src/io_v2/base_writer/position_delete_writer.rs

```diff
@@ -102,6 +118,172 @@ impl Combinable for PositionDeleteInput {
         1
     }
 }
+#[cfg(feature = "prometheus")]
+mod prometheus {
+    use crate::io_v2::{FileWriterBuilder, IcebergWriter, IcebergWriterBuilder};
+    use crate::Result;
+    use prometheus::core::{AtomicU64, GenericGauge};
+
+    use super::{
+        PositionDeleteInput, PositionDeleteMetrics, PositionDeleteWriter,
+        PositionDeleteWriterBuilder,
+    };
+
+    #[derive(Clone)]
+    pub struct PositionDeleteWriterWithMetricsBuilder<B: FileWriterBuilder> {
+        current_cache_number: GenericGauge<AtomicU64>,
+        inner: PositionDeleteWriterBuilder<B>,
+    }
+
+    impl<B: FileWriterBuilder> PositionDeleteWriterWithMetricsBuilder<B> {
+        pub fn new(
+            inner: PositionDeleteWriterBuilder<B>,
+            current_cache_number: GenericGauge<AtomicU64>,
+        ) -> Self {
+            Self {
+                inner,
+                current_cache_number,
+            }
+        }
+    }
+
+    #[async_trait::async_trait]
+    impl<B: FileWriterBuilder> IcebergWriterBuilder for PositionDeleteWriterWithMetricsBuilder<B> {
+        type R = PositionDeleteWriterWithMetrics<B>;
+
+        async fn build(self, schema: &arrow_schema::SchemaRef) -> crate::Result<Self::R> {
+            let writer = self.inner.build(schema).await?;
+            Ok(PositionDeleteWriterWithMetrics {
+                writer,
+                cache_number: self.current_cache_number,
+                current_metrics: PositionDeleteMetrics {
+                    current_cache_number: 0,
+                },
+            })
+        }
+    }
+
+    pub struct PositionDeleteWriterWithMetrics<B: FileWriterBuilder> {
+        writer: PositionDeleteWriter<B>,
+
+        // metrics
+        cache_number: GenericGauge<AtomicU64>,
+        current_metrics: PositionDeleteMetrics,
+    }
+
+    impl<B: FileWriterBuilder> PositionDeleteWriterWithMetrics<B> {
+        fn update_metrics(&mut self) -> Result<()> {
+            let last_metrics = std::mem::replace(&mut self.current_metrics, self.writer.metrics());
+            {
+                let delta = self.current_metrics.current_cache_number as i64
+                    - last_metrics.current_cache_number as i64;
+                if delta > 0 {
+                    self.cache_number.add(delta as u64);
```

Comment:
For a guage, it will be dispatch to multiple writer by a builder. So the gauge is the sum of multiple writers, that's why we can't set it value directly.

Reply:
Got it, thanks!